mockgcp/mockcompute/nodegroupsv1.go

```diff
@@ -0,0 +1,183 @@
+)
+
+type NodeGroupsV1 struct {
+	*MockService
+	pb.UnimplementedNodeGroupsServer
+}
+
+func (s *NodeGroupsV1) Get(ctx context.Context, req *pb.GetNodeGroupRequest) (*pb.NodeGroup, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeGroup{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node group %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading node group: %v", err)
+		}
+	}
+
+	return obj, nil
+}
+
+func (s *NodeGroupsV1) Insert(ctx context.Context, req *pb.InsertNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroupResource().GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	id := s.generateID()
+
+	obj := proto.Clone(req.GetNodeGroupResource()).(*pb.NodeGroup)
+	obj.SelfLink = PtrTo("https://compute.googleapis.com/compute/v1/" + name.String())
+	obj.CreationTimestamp = PtrTo(s.nowString())
+	obj.Id = &id
+	obj.Kind = PtrTo("compute#nodegroup")
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating node group: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) Patch(ctx context.Context, req *pb.PatchNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.NodeGroup{}
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating node group: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) Delete(ctx context.Context, req *pb.DeleteNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeGroup{}
+	if err := s.storage.Delete(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node group %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting node group: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) SetNodeTemplate(ctx context.Context, req *pb.SetNodeTemplateNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeTemplateNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup(), req.GetNodeGroupsSetNodeTemplateRequestResource().GetNodeTemplate())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeGroup{}
```

Comment:
I'm unsure if `setNodeTemplate` is necessary now. 
I tried to remove it and the mock test still passed without issues or message like "setNodeTemplate is not implemented". However, I remember I got the `not supported` error once so that's why I implemented this method.

Reply:
I have decided to temporarily remove "setNodeTemplate" because I don't see any issues.

Reply:
sgtm - if we're not using it in the tests, probably simpler just to drop it until we can verify it against a test!